### DISCUSSION TITLE
Emails are stuck in the queue as pending since twodays

### DISCUSSION THREAD

**Discussion by XenyFord** [postalserver/postal#1999] [Development discussions]
Hi Friends,

Can anyone help me on how to solve this problem as I'm unable to send emails that got stuck in the queue as pending, though I made lot of restarts since yesterday, ans also have 4GB memory available free.

Please help me If there is a quick solution to send the emails immediately? 

Thank you in advance!

**Comment by willpower232** [comment]
What is your `postal status`? Is everything up?

**Reply by XenyFord** [reply]
Postal status is showing UP.

**Reply by casualmatt** [reply]
try to restart, it's happened to me too

**Reply by XenyFord** [reply]
Restarted many times. But still no use.

**Reply by willpower232** [reply]
is every part of Postal up though?

**Reply by XenyFord** [reply]
Yes sir

**Reply by XenyFord** [reply]
Can u please advise me what to do at this stage?

**Reply by XenyFord** [reply]
Also, Can anyone help me on how to release the pending email in the queue?

**Reply by casualmatt** [reply]
restart didn't work, did you check if the host that you want to deliver those mails are in the suppressions list? 
**messages>suppressions**.

**Reply by XenyFord** [reply]
There are only 300 emails in the suppression list, but more than 10K emails are still in the queue as pending since several weeks ago...

**Reply by casualmatt** [reply]
is possible that your provider locks your port on the server, what's the error if you manually try to send one of those mail.

**Reply by willpower232** [reply]
if the email is pending then either the worker is not running or you have set up the wrong IP address in config or pool.

If the provider was blocking the email then the email would say it couldn't be delivered, not that it was pending.

If the suppression list was blocking the email then it would be held not pending.

**Reply by XenyFord** [reply]
The pending emails are shown as in the image below..
![Screenshot_1](https://user-images.githubusercontent.com/89290310/172193140-045bdeba-3a59-4f25-8839-e01640b5d9a0.jpg)
.

**Reply by XenyFord** [reply]
Hi Mr. Will,

Also, please be very kind to help me on how to solve this. But, my humble request to you to please try write a bit elaborated here so everyone can easily understand, because maximum number of the members here are technically not well sound. 
Please don't get me wrong in any manner. 

Thank you in advance!

**Reply by willpower232** [reply]
> if the email is pending then either the worker is not running or you have set up the wrong IP address in config or pool.

**Reply by XenyFord** [reply]
I have only one IP correctly configured. Worker is also running right way, because I sent few other TEST emails today to different email addresses are delivered, but not the emails that are still in the queue.

**Reply by XenyFord** [reply]
Also, please let me know the exact command to remove or empty all the pending emails to give a fresh start again.

**Reply by willpower232** [reply]
Is it only much older messages which are pending? You can see in the screenshot that the raw message no longer exists which means there isn't a message to send. It is possible you were having issues several weeks ago which are not present any longer.

The only way to clear it would be to go in your database and edit or remove the rows.

**Reply by abdallahtaleb** [reply]
I am having the exact same issue. Only 1 message is being sent after I restart Postal then everything goes to queue. Can anyone help?

**Comment by willpower232** [comment]
@abdallahtaleb you can see in the other thread that you need to check `postal status` to make sure everything is up then

> if the email is pending then either the worker is not running or you have set up the wrong IP address in config or pool.
